visualization/app/codeCharta/codeCharta.service.ts

```diff
@@ -52,36 +69,38 @@ export class CodeChartaService {
 		}
 	}
 
-	private getAttributeTypes(attributeTypes: AttributeTypes): AttributeTypes {
-		const newAttributeTypes: any = {}
-
-		if (_.isEmpty(attributeTypes) || !attributeTypes) {
+	private getAttributeTypes(attributeTypes: AttributeTypes | OldAttributeTypes): AttributeTypes {
+		if (_.isEmpty(attributeTypes) || !attributeTypes || _.isArray(attributeTypes.nodes) || _.isArray(attributeTypes.edges)) {
 			return {
 				nodes: {},
 				edges: {}
 			}
-		} else {
-			if (!attributeTypes.nodes) {
-				newAttributeTypes.nodes = {}
-			} else {
-				newAttributeTypes.nodes = attributeTypes.nodes
-			}
+		}
 
-			if (!attributeTypes.edges) {
-				newAttributeTypes.edges = {}
-			} else {
-				newAttributeTypes.edges = attributeTypes.edges
-			}
+		return {
+			nodes: !attributeTypes.nodes ? {} : attributeTypes.nodes,
+			edges: !attributeTypes.edges ? {} : attributeTypes.edges
 		}
-		return newAttributeTypes
 	}
 
 	private potentiallyUpdateBlacklistTypes(blacklist): BlacklistItem[] {
 		blacklist.forEach(x => {
-			if (x.type === "hide") {
+			if (x.type === ExportBlacklistType.hide) {
 				x.type = BlacklistType.flatten
 			}
 		})
 		return blacklist
 	}
+
+	private setDefaultScenario() {
+		const metricData = this.metricService.getMetricData()
+		const { areaMetric, heightMetric, colorMetric } = ScenarioHelper.getDefaultScenarioSetting().dynamicSettings
+		const set = new Set([areaMetric, heightMetric, colorMetric])
+
+		metricData.forEach(x => set.delete(x.name))
```

Comment:
We have to find out, if we can actually apply the default scenarios. When we don't have rloc or mcc available in the cc.json we can't.
The set adds these metric names in a set (resulting in a set containing ["rloc", "mcc"]. If we encounter that metric name in our metricData from the cc.json, we can remove it from the set.
If the set is empty, the scenario can be applied.

Reply:
Your suggestion wasn't correct. Instead of checking if every metric in metricData exists in the metrics from the scenario, we have to actually check if the metrics from the scenario exists in the metricData. :)